torchtext/csrc/register_pybindings.cpp

```diff
@@ -179,6 +179,16 @@ PYBIND11_MODULE(_torchtext, m) {
       .def_property_readonly("byte_encoder_", &GPT2BPEEncoder::GetByteEncoder)
       .def("encode", &GPT2BPEEncoder::Encode)
       .def("tokenize", &GPT2BPEEncoder::Tokenize)
+      .def(
+          "add_special_tokens",
+          [](const c10::intrusive_ptr<GPT2BPEEncoder>& self,
+             const std::unordered_map<std::string, std::string>& items,
+             const std::vector<std::string>& additional) {
+            c10::Dict<std::string, std::string> d;
+            for (const auto& item : items)
+              d.insert(item.first, item.second);
+            return (self->AddSpecialTokens(d, additional));
+          })
```

Comment:
I originally tried using the `std::unordered_map`, but I get the following compilation error (`unsupported input type: std::unordered_map<Key, Value>. Please use Dict<Key, Value> instead.`) on doing so.

```
/Users/sumitkumar/mambaforge/envs/bootcamp-rl/lib/python3.10/site-packages/torch/include/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h:164:5: error: static_assert failed "You tried to register a kernel with an unsupported input type: std::unordered_map<Key, Value>. Please use Dict<Key, Value> instead."
    static_assert(AllowDeprecatedTypes,
    ^             ~~~~~~~~~~~~~~~~~~~~
```

So, I switched to `c10::Dict` instead. But it turned out that the Python dictionary doesn't directly convert to our custom `c10:Dict` map, and it gives me following error if I try to do so.

```
TypeError: add_special_tokens(): incompatible function arguments. The following argument types are supported:
    1. (self: torchtext._torchtext.GPT2BPEEncoder, arg0: c10::Dict<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, long long>) -> int

Invoked with: <torchtext._torchtext.GPT2BPEEncoder object at 0x12428de70>, {}

Did you forget to `#include <pybind11/stl.h>`? Or <pybind11/complex.h>,
<pybind11/functional.h>, <pybind11/chrono.h>, etc. Some automatic
conversions are optional and require extra headers to be included
when compiling your pybind11 module.
```

So, I chose to pass Python `dict` to C++ `unordered_map`, but convert it bindings to avoid the first error. This is similar to the logic for constructor.